botocore/credentials.py

```diff
@@ -550,19 +554,51 @@ def load(self):
         """
         if self._mapping['access_key'] in self.environ:
             logger.info('Found credentials in environment variables.')
-            access_key, secret_key = self._extract_creds_from_mapping(
-                self.environ, self._mapping['access_key'],
-                self._mapping['secret_key'])
-            token = self._get_session_token()
-            return Credentials(access_key, secret_key, token,
-                               method=self.METHOD)
+            fetcher = self._create_credentials_fetcher()
+            credentials = fetcher()
+            expiry_time = credentials['expiry_time']
+            if expiry_time is not None:
```

Comment:
Yeah I cannot think of other ways to do this. I guess another question I have is it possible to transition from refershable environment credentials to non-refreshable? So say I had an expiry time set and then unset the expiry time variable before the next refresh what would happen?